File: spec/javascripts/dataStructure/PriorityQueueSpec.js
//Array.prototype.join = (function(originalJoin) {
//  return function(separator) {
//    return originalJoin.call(this, separator===undefined?separator:'|');
//  }; 
//})(Array.prototype.join);

describe('PriorityQueue', function() {

  describe('enqueue', function() {
    it('should sort input values based on there priority', function() {
      pq = new PriorityQueue(10);

      pq.enqueue(7);
      expect(pq.front()).toEqual(7);
      expect(pq.rear()).toEqual(7);
      pq.enqueue(6);
      expect(pq.front()).toEqual(6)
      expect(pq.rear()).toEqual(7);
      pq.enqueue(5);
      expect(pq.front()).toEqual(5)
      expect(pq.rear()).toEqual(7);
    });
  });
});

File: spec/Priority_Queue_Spec.rb
require 'spec_helper'

describe 'Priority_Queue' do

  describe 'enqueue' do
    it 'should sort input values based on there priority' do
      pq = Priority_Queue.new 10

      pq.enqueue 10
      pq.front.should eq 10

      pq.enqueue 9
      pq.front.should eq 9
      pq.rear.should eq 10
    end
  end
end
